services/cd-service/pkg/cmd/server.go

```diff
@@ -336,39 +336,47 @@ func RunServer() {
 			}
 			getAllReleases := func(ctx context.Context, app string) (db.AllReleases, error) {
 				s := repo.State()
-				releases, err := s.GetApplicationReleases(app)
+				releases, err := s.GetAllApplicationReleasesFromFile(app)
 				if err != nil {
 					return nil, fmt.Errorf("cannot get releases of app %s: %v", app, err)
 				}
 				var result = db.AllReleases{}
 				for i := range releases {
 					releaseVersion := releases[i]
-					repoRelease, err := s.GetApplicationRelease(app, releaseVersion)
+					repoRelease, err := s.GetApplicationReleaseFromManifest(app, releaseVersion)
 					if err != nil {
 						return nil, fmt.Errorf("cannot get app release of app %s and release %v: %v", app, releaseVersion, err)
 					}
-					manifests, err := s.GetApplicationReleaseManifests(app, releaseVersion)
+					manifests, err := s.GetApplicationReleaseManifestsFromFile(app, releaseVersion)
 					if err != nil {
 						return nil, fmt.Errorf("cannot get manifest for app %s and release %v: %v", app, releaseVersion, err)
 					}
+					var manifestsMap = map[string]string{}
 					for index := range manifests {
 						manifest := manifests[index]
-						result[releaseVersion] = db.ReleaseWithManifest{
-							Version:         releaseVersion,
-							UndeployVersion: repoRelease.UndeployVersion,
-							SourceAuthor:    repoRelease.SourceAuthor,
-							SourceCommitId:  repoRelease.SourceCommitId,
-							SourceMessage:   repoRelease.SourceMessage,
-							CreatedAt:       repoRelease.CreatedAt,
-							DisplayVersion:  repoRelease.DisplayVersion,
-							Manifest:        manifest.Content,
-							Environment:     manifest.Environment,
-						}
+						manifestsMap[manifest.Environment] = manifest.Content
+					}
+					result[releaseVersion] = db.ReleaseWithManifest{
+						Version:         releaseVersion,
+						UndeployVersion: repoRelease.UndeployVersion,
+						SourceAuthor:    repoRelease.SourceAuthor,
+						SourceCommitId:  repoRelease.SourceCommitId,
+						SourceMessage:   repoRelease.SourceMessage,
+						CreatedAt:       repoRelease.CreatedAt,
+						DisplayVersion:  repoRelease.DisplayVersion,
+						Manifests:       manifestsMap,
 					}
 				}
 				return result, nil
 			}
-			migErr := dbHandler.RunCustomMigrations(ctx, getAppsAndTeams, repo.State().GetCurrentlyDeployed, getAllReleases, repo.State().GetCurrentEnvironmentLocks, repo.State().GetCurrentApplicationLocks)
+			migErr := dbHandler.RunCustomMigrations(
+				ctx,
+				getAppsAndTeams,
```

Comment:
I don't see why `getAppsAndTeams` and `getAllReleases` are different from other callbacks passed to `RunCustomMigrations`, in my understanding, they all have the same "nature". If there is indeed nothing particular about them, I suggest to reflect that in this function call here.

Reply:
I made them all the same